Modules/_io/bufferedio.c

```diff
@@ -819,6 +821,60 @@ _buffered_raw_seek(buffered *self, Py_off_t target, int whence)
     return n;
 }
 
+#ifdef HAVE_FORK
+
+static PyObject*
+buffered_after_fork_child_impl(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    buffered *buf = (buffered *)self;
+    if (_buffered_init(buf) < 0) {
+        PyErr_SetString(PyExc_RuntimeError, "Failed to initialize buffer after fork");
+        return NULL;
+    }
+    _bufferedreader_reset_buf(buf);
+    Py_RETURN_NONE;
+}
+
+static PyMethodDef buffered_fork_methods[] = {
+    {"_after_fork_child", buffered_after_fork_child_impl, METH_NOARGS, NULL},
+    {NULL, NULL}
+};
+
+static int
+buffered_register_at_fork(buffered *self)
+{
+    PyInterpreterState *interp = PyThreadState_Get()->interp;
+    if (!interp) {
+        PyErr_SetString(PyExc_RuntimeError, "Failed to get interpreter state");
+        return -1;
+    }
+
+    // Only register the fork handlers once
+    if (!interp->after_forkers_child) {
+        interp->after_forkers_child = PyList_New(0);
+        if (!interp->after_forkers_child) {
+            return -1;
+        }
+    }
+
+    /* Create method objects */
+    PyObject *after_child = PyCFunction_New(&buffered_fork_methods[0], (PyObject *)self);
+    if (!after_child) {
+        return -1;
+    }
+
+    /* Append callbacks to the lists */
+    int status = 0;
+    if (PyList_Append(interp->after_forkers_child, after_child) < 0) {
```

Comment:
If I'm correct, I think the length of `after_forkers_child` will raising unlimited？